src/openvic-simulation/misc/Modifier.cpp

```diff
@@ -291,23 +303,35 @@ bool ModifierManager::load_triggered_modifiers(ast::NodeCPtr root) {
 key_value_callback_t ModifierManager::_modifier_effect_callback(
 	ModifierValue& modifier, key_value_callback_t default_callback, ModifierEffectValidator auto effect_validator
 ) const {
-
-	return [this, &modifier, default_callback, effect_validator](std::string_view key, ast::NodeCPtr value) -> bool {
-		ModifierEffect const* effect = get_modifier_effect_by_identifier(key);
-		if (effect != nullptr) {
-			if (effect_validator(*effect)) {
-				if (!modifier.values.contains(effect)) {
-					return expect_fixed_point(assign_variable_callback(modifier.values[effect]))(value);
-				} else {
-					Logger::error("Duplicate modifier effect: ", key);
-					return false;
-				}
+	std::function<bool(ModifierEffect const*, ast::NodeCPtr)> add_modifier_cb = [this, &modifier, effect_validator](ModifierEffect const* effect, ast::NodeCPtr value) -> bool {
+		if (effect_validator(*effect)) {
+			if (!modifier.values.contains(effect)) {
+				return expect_fixed_point(assign_variable_callback(modifier.values[effect]))(value);
 			} else {
-				Logger::error("Failed to validate modifier effect: ", key);
+				Logger::error("Duplicate modifier effect: ", effect->get_identifier());
 				return false;
 			}
+		} else {
+			Logger::error("Failed to validate modifier effect: ", effect->get_identifier());
+			return false;
 		}
-		return default_callback(key, value);
+	};
+
+	return [this, &modifier, default_callback, effect_validator, add_modifier_cb](std::string_view key, ast::NodeCPtr value) -> bool {
+		ModifierEffect const* effect = get_modifier_effect_by_identifier(key);
+		if (effect != nullptr && value->cast_to<ast::IdentifierNode>() != nullptr) {
+			return add_modifier_cb(effect, value);
+		} else if (complex_modifiers.contains(key) && value->cast_to<ast::AbstractListNode>() != nullptr) {
```

Comment:
```suggestion
		} else if (complex_modifiers.contains(key) && value->is_derived_from<ast::AbstractListNode>()) {
```
Again, clever solution to a nasty problem, this time `artillery` being used as a good to be encouraged by a national focus, and not as the key for a list of modifiers to the military unit of the same name.

Eventually (in a future PR, this one is massive enough as is 😁), we're going to need to design a proper system for handling these multiple use keys. Some ideas:
- similar to `default_callback`, have a function that's called at the beginning and only parse modifier effects if that function doesn't accept the key-value pair (the function would need to return 3 values, success/failure for keys it accepts and a third value for keys it rejects that need to be parsed as modifier effects).
- change how effect validators are used so that modifier keys they reject go to `default_callback`, instead of being thrown away as "failed keys". This requires less code change but more work to actually use, as it requires listing every single allowed modifier effect while the first idea just moves some of the already existing default parsing code to the beginning of the process.

Again, let's leave this for now, it needs some serious brainstorming and research into which modifier effect keys also have special uses.